### PR TITLE
API-10008: Document suggested behavior on RTM reconnect

### PR DESCRIPTION
**Pull request by grz-gajda**
# 🚀 Links

- [Jira](https://livechatinc.atlassian.net/browse/API-10008)

### CODE REVIEW

**Review by bkubiak — APPROVED** [review]

File: src/pages/messaging/customer-chat-api/v3.4/rtm-pushes/index.mdx [hunk 1]
```diff
@@ -519,22 +519,22 @@ Informs that a Customer was disconnected. The payload contains the reason of Cus
...
-| `unsupported_version`                | Connecting to an unsupported version of the Customer Chat API.                                                                       |
+Based on received `reason` we suggest different reaction.
+
+| Type                                 | Notes                                                                                                                                | Suggested behavior                              |
+| ------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------ | ----------------------------------------------- |
+| `access_token_expired`               | Access token lifetime has elapsed.                                                                                                   | Reconnect. Please regenerate your access token. |
+| `connection_evicted`                 | The Customer Chat API should be available again soon; reconnecting is advised                                                        | Reconnect.                                      |
```
Comment (bkubiak) [inline src/pages/messaging/customer-chat-api/v3.4/rtm-pushes/index.mdx:527]: > reconnecting is advised

perhaps this is no longer needed

File: src/pages/messaging/customer-chat-api/v3.4/rtm-pushes/index.mdx [hunk 2]
```diff
@@ -519,22 +519,22 @@ Informs that a Customer was disconnected. The payload contains the reason of Cus
...
+| `license_expired`                    | The license has expired.                                                                                                             | Don't reconnect.                                |
+| `misdirected_connection` **\*\***    | Customer connected to a server in the wrong region.                                                                                  | Don't reconnect.                                |
+| `unsupported_version`                | Connecting to an unsupported version of the Customer Chat API.                                                                       | Don't reconnect.                                |
+| `product_version_changed`            | The product version has changed.                                                                                                     | Don't reconnect.                                |
+| `service_temporarily_unavailable`    | The Customer Chat API should be available again soon; reconnecting is advised                                                        | Reconnect.                                      |
```
Comment (bkubiak) [inline src/pages/messaging/customer-chat-api/v3.4/rtm-pushes/index.mdx:535]: > reconnecting is advised

perhaps this is no longer needed

**Review by bkubiak — APPROVED** [review]

**Comment by bkubiak** [comment]
Please remember to align with https://github.com/livechat/livechat-public-docs/pull/1083 if it is merged first.

Also please pay attention to the new version v3.5: https://livechatinc.atlassian.net/browse/DPS-3314.

File: src/pages/messaging/agent-chat-api/rtm-pushes/index.mdx [hunk 3]
```diff
@@ -692,22 +692,22 @@ Informs that an Agent was disconnected. The payload contains the reason of Custo
...
+| `agent_disconnected_by_server`      | Agent has been disconnected by the server.                                                                                           | Don't reconnect.                                |
+| `access_token_revoked`              | Agent access token has been revoked.                                                                                                 | Don't reconnect.                                |
+| `access_token_expired`              | Access token life time has elapsed.                                                                                                  | Reconnect. Please regenerate your access token. |
+| `connection_timeout`                | Has not received [`ping`](#authentication) from the client for some time, or it's been too long since the connection was authorized. | Reconnect.                                      |
+| `internal_error`                    | Internal error                                                                                                                       | Reconnect.                                      |
```
Comment (oliwiapolec) [inline src/pages/messaging/agent-chat-api/rtm-pushes/index.mdx:703]:
```suggestion
| `internal_error`                    | Internal error.                                                                                                                       | Reconnect.                                      |
```

File: src/pages/messaging/agent-chat-api/v3.1/rtm-pushes/index.mdx [hunk 4]
```diff
@@ -690,21 +690,22 @@ Informs that an Agent was disconnected. The payload contains the reason of Custo
...
+| `agent_disconnected_by_server`      | Agent has been disconnected by the server.                                                                                           | Don't reconnect.                                |
+| `access_token_revoked`              | Agent access token has been revoked.                                                                                                 | Don't reconnect.                                |
+| `access_token_expired`              | Access token life time has elapsed.                                                                                                  | Reconnect. Please regenerate your access token. |
+| `connection_timeout`                | Has not received [`ping`](#authentication) from the client for some time, or it's been too long since the connection was authorized. | Reconnect.                                      |
+| `internal_error`                    | Internal error                                                                                                                       | Reconnect.                                      |
```
Comment (oliwiapolec) [inline src/pages/messaging/agent-chat-api/v3.1/rtm-pushes/index.mdx:701]:
```suggestion
| `internal_error`                    | Internal error.                                                                                                                       | Reconnect.                                      |
```

File: src/pages/messaging/agent-chat-api/v3.2/rtm-pushes/index.mdx [hunk 5]
```diff
@@ -698,21 +698,22 @@ Informs that an Agent was disconnected. The payload contains the reason of Custo
...
+| `agent_disconnected_by_server`      | Agent has been disconnected by the server.                                                                                           | Don't reconnect.                                |
+| `access_token_revoked`              | Agent access token has been revoked.                                                                                                 | Don't reconnect.                                |
+| `access_token_expired`              | Access token life time has elapsed.                                                                                                  | Reconnect. Please regenerate your access token. |
+| `connection_timeout`                | Has not received [`ping`](#authentication) from the client for some time, or it's been too long since the connection was authorized. | Reconnect.                                      |
+| `internal_error`                    | Internal error                                                                                                                       | Reconnect.                                      |
```
Comment (oliwiapolec) [inline src/pages/messaging/agent-chat-api/v3.2/rtm-pushes/index.mdx:709]:
```suggestion
| `internal_error`                    | Internal error.                                                                                                                       | Reconnect.                                      |
```

File: src/pages/messaging/agent-chat-api/v3.4/rtm-pushes/index.mdx [hunk 6]
```diff
@@ -714,21 +714,23 @@ Informs that an Agent was disconnected. The payload contains the reason of Custo
...
+| `access_token_revoked`              | Agent access token has been revoked.                                                                                                   | Don't reconnect.                                |
+| `connection_evicted`                | Sent to the connection that's replaced by a newly established one, and the new connection exceeded the lim it of possible connections. | Don't reconnect.                                |
+| `access_token_expired`              | Access token life time has elapsed.                                                                                                    | Reconnect. Please regenerate your access token. |
+| `connection_timeout`                | Has not received [`ping`](#authentication) from the client for some time, or it's been too long since the connection was authorized.   | Reconnect.                                      |
+| `internal_error`                    | Internal error                                                                                                                         | Reconnect.                                      |
```
Comment (oliwiapolec) [inline src/pages/messaging/agent-chat-api/v3.4/rtm-pushes/index.mdx:726]:
```suggestion
| `internal_error`                    | Internal error.                                                                                                                         | Reconnect.                                      |
```

File: src/pages/messaging/customer-chat-api/v3.4/rtm-pushes/index.mdx [hunk 7]
```diff
@@ -519,22 +519,22 @@ Informs that a Customer was disconnected. The payload contains the reason of Cus
...
+| ------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------ | ----------------------------------------------- |
+| `access_token_expired`               | Access token lifetime has elapsed.                                                                                                   | Reconnect. Please regenerate your access token. |
+| `connection_evicted`                 | The Customer Chat API should be available again soon;                                                                                | Reconnect.                                      |
+| `connection_timeout`                 | Has not received [`ping`](#authentication) from the client for some time, or it's been too long since the connection was authorized. | Reconnect.                                      |
+| `internal_error`                     | Internal, unexpected error;                                                                                                          | Reconnect.                                      |
```
Comment (oliwiapolec) [inline src/pages/messaging/customer-chat-api/v3.4/rtm-pushes/index.mdx:529]:
```suggestion
| `internal_error`                     | Internal error.                                                                                                          | Reconnect.                                      |
```

File: src/pages/messaging/customer-chat-api/rtm-pushes/index.mdx [hunk 8]
```diff
@@ -519,22 +519,22 @@ Informs that a Customer was disconnected. The payload contains the reason of Cus
...
+| ------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------ | ----------------------------------------------- |
+| `access_token_expired`               | Access token lifetime has elapsed.                                                                                                   | Reconnect. Please regenerate your access token. |
+| `connection_evicted`                 | The Customer Chat API should be available again soon;                                                                                | Reconnect.                                      |
+| `connection_timeout`                 | Has not received [`ping`](#authentication) from the client for some time, or it's been too long since the connection was authorized. | Reconnect.                                      |
+| `internal_error`                     | Internal, unexpected error;                                                                                                          | Reconnect.                                      |
```
Comment (oliwiapolec) [inline src/pages/messaging/customer-chat-api/rtm-pushes/index.mdx:529]:
```suggestion
| `internal_error`                     | Internal error.                                                                                                          | Reconnect.                                      |
```

File: src/pages/messaging/customer-chat-api/v3.1/rtm-pushes/index.mdx [hunk 9]
```diff
@@ -529,22 +529,22 @@ Informs that a Customer was disconnected. The payload contains the reason of Cus
...
+| ------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------ | ----------------------------------------------- |
+| `access_token_expired`               | Access token lifetime has elapsed.                                                                                                   | Reconnect. Please regenerate your access token. |
+| `connection_evicted`                 | The Customer Chat API should be available again soon;                                                                                | Reconnect.                                      |
+| `connection_timeout`                 | Has not received [`ping`](#authentication) from the client for some time, or it's been too long since the connection was authorized. | Reconnect.                                      |
+| `internal_error`                     | Internal, unexpected error;                                                                                                          | Reconnect.                                      |
```
Comment (oliwiapolec) [inline src/pages/messaging/customer-chat-api/v3.1/rtm-pushes/index.mdx:539]:
```suggestion
| `internal_error`                     | Internal error.                                                                                                          | Reconnect.                                      |
```

File: src/pages/messaging/customer-chat-api/v3.2/rtm-pushes/index.mdx [hunk 10]
```diff
@@ -520,22 +520,22 @@ Informs that a Customer was disconnected. The payload contains the reason of Cus
...
+| ------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------ | ----------------------------------------------- |
+| `access_token_expired`               | Access token lifetime has elapsed.                                                                                                   | Reconnect. Please regenerate your access token. |
+| `connection_evicted`                 | The Customer Chat API should be available again soon;                                                                                | Reconnect.                                      |
+| `connection_timeout`                 | Has not received [`ping`](#authentication) from the client for some time, or it's been too long since the connection was authorized. | Reconnect.                                      |
+| `internal_error`                     | Internal, unexpected error;                                                                                                          | Reconnect.                                      |
```
Comment (oliwiapolec) [inline src/pages/messaging/customer-chat-api/v3.2/rtm-pushes/index.mdx:530]:
```suggestion
| `internal_error`                     | Internal error.                                                                                                         | Reconnect.                                      |
```

File: src/pages/messaging/customer-chat-api/rtm-pushes/index.mdx [hunk 11]
```diff
@@ -519,22 +519,22 @@ Informs that a Customer was disconnected. The payload contains the reason of Cus
...
-| `unsupported_version`                | Connecting to an unsupported version of the Customer Chat API.                                                                       |
+Based on received `reason` we suggest different reaction.
+
+| Type                                 | Notes                                                                                                                                | Suggested behavior                              |
+| ------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------ | ----------------------------------------------- |
+| `access_token_expired`               | Access token lifetime has elapsed.                                                                                                   | Reconnect. Please regenerate your access token. |
+| `connection_evicted`                 | The Customer Chat API should be available again soon;                                                                                | Reconnect.                                      |
```
Comment (oliwiapolec) [inline src/pages/messaging/customer-chat-api/rtm-pushes/index.mdx:527]:
```suggestion
| `connection_evicted`                 | The Customer Chat API should be available again soon.                                                                                | Reconnect.                                      |
```

File: src/pages/messaging/customer-chat-api/rtm-pushes/index.mdx [hunk 12]
```diff
@@ -519,22 +519,22 @@ Informs that a Customer was disconnected. The payload contains the reason of Cus
...
+| `license_expired`                    | The license has expired.                                                                                                             | Don't reconnect.                                |
+| `misdirected_connection` **\*\***    | Customer connected to a server in the wrong region.                                                                                  | Don't reconnect.                                |
+| `unsupported_version`                | Connecting to an unsupported version of the Customer Chat API.                                                                       | Don't reconnect.                                |
+| `product_version_changed`            | The product version has changed.                                                                                                     | Don't reconnect.                                |
+| `service_temporarily_unavailable`    | The Customer Chat API should be available again soon;                                                                                | Reconnect.                                      |
```
Comment (oliwiapolec) [inline src/pages/messaging/customer-chat-api/rtm-pushes/index.mdx:535]:
```suggestion
| `service_temporarily_unavailable`    | The Customer Chat API should be available again soon.                                                                                | Reconnect.                                      |
```

File: src/pages/messaging/customer-chat-api/v3.1/rtm-pushes/index.mdx [hunk 13]
```diff
@@ -529,22 +529,22 @@ Informs that a Customer was disconnected. The payload contains the reason of Cus
...
-| `unsupported_version`                | Connecting to an unsupported version of the Customer Chat API.                                                                       |
+Based on received `reason` we suggest different reaction.
+
+| Type                                 | Notes                                                                                                                                | Suggested behavior                              |
+| ------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------ | ----------------------------------------------- |
+| `access_token_expired`               | Access token lifetime has elapsed.                                                                                                   | Reconnect. Please regenerate your access token. |
+| `connection_evicted`                 | The Customer Chat API should be available again soon;                                                                                | Reconnect.                                      |
```
Comment (oliwiapolec) [inline src/pages/messaging/customer-chat-api/v3.1/rtm-pushes/index.mdx:537]:
```suggestion
| `connection_evicted`                 | The Customer Chat API should be available again soon.                                                                                | Reconnect.                                      |
```

File: src/pages/messaging/customer-chat-api/v3.1/rtm-pushes/index.mdx [hunk 14]
```diff
@@ -529,22 +529,22 @@ Informs that a Customer was disconnected. The payload contains the reason of Cus
...
+| `license_expired`                    | The license has expired.                                                                                                             | Don't reconnect.                                |
+| `misdirected_connection` **\*\***    | Customer connected to a server in the wrong region.                                                                                  | Don't reconnect.                                |
+| `unsupported_version`                | Connecting to an unsupported version of the Customer Chat API.                                                                       | Don't reconnect.                                |
+| `product_version_changed`            | The product version has changed.                                                                                                     | Don't reconnect.                                |
+| `service_temporarily_unavailable`    | The Customer Chat API should be available again soon;                                                                                | Reconnect.                                      |
```
Comment (oliwiapolec) [inline src/pages/messaging/customer-chat-api/v3.1/rtm-pushes/index.mdx:545]:
```suggestion
| `service_temporarily_unavailable`    | The Customer Chat API should be available again soon.                                                                                | Reconnect.                                      |
```

File: src/pages/messaging/customer-chat-api/v3.2/rtm-pushes/index.mdx [hunk 15]
```diff
@@ -520,22 +520,22 @@ Informs that a Customer was disconnected. The payload contains the reason of Cus
...
-| `unsupported_version`                | Connecting to an unsupported version of the Customer Chat API.                                                                       |
+Based on received `reason` we suggest different reaction.
+
+| Type                                 | Notes                                                                                                                                | Suggested behavior                              |
+| ------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------ | ----------------------------------------------- |
+| `access_token_expired`               | Access token lifetime has elapsed.                                                                                                   | Reconnect. Please regenerate your access token. |
+| `connection_evicted`                 | The Customer Chat API should be available again soon;                                                                                | Reconnect.                                      |
```
Comment (oliwiapolec) [inline src/pages/messaging/customer-chat-api/v3.2/rtm-pushes/index.mdx:528]:
```suggestion
| `connection_evicted`                 | The Customer Chat API should be available again soon.                                                                                | Reconnect.                                      |
```

File: src/pages/messaging/customer-chat-api/v3.2/rtm-pushes/index.mdx [hunk 16]
```diff
@@ -520,22 +520,22 @@ Informs that a Customer was disconnected. The payload contains the reason of Cus
...
+| `license_expired`                    | The license has expired.                                                                                                             | Don't reconnect.                                |
+| `misdirected_connection` **\*\***    | Customer connected to a server in the wrong region.                                                                                  | Don't reconnect.                                |
+| `unsupported_version`                | Connecting to an unsupported version of the Customer Chat API.                                                                       | Don't reconnect.                                |
+| `product_version_changed`            | The product version has changed.                                                                                                     | Don't reconnect.                                |
+| `service_temporarily_unavailable`    | The Customer Chat API should be available again soon;                                                                                | Reconnect.                                      |
```
Comment (oliwiapolec) [inline src/pages/messaging/customer-chat-api/v3.2/rtm-pushes/index.mdx:536]:
```suggestion
| `service_temporarily_unavailable`    | The Customer Chat API should be available again soon.                                                                                | Reconnect.                                      |
```

File: src/pages/messaging/customer-chat-api/v3.4/rtm-pushes/index.mdx [hunk 17]
```diff
@@ -519,22 +519,22 @@ Informs that a Customer was disconnected. The payload contains the reason of Cus
...
-| `unsupported_version`                | Connecting to an unsupported version of the Customer Chat API.                                                                       |
+Based on received `reason` we suggest different reaction.
+
+| Type                                 | Notes                                                                                                                                | Suggested behavior                              |
+| ------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------ | ----------------------------------------------- |
+| `access_token_expired`               | Access token lifetime has elapsed.                                                                                                   | Reconnect. Please regenerate your access token. |
+| `connection_evicted`                 | The Customer Chat API should be available again soon;                                                                                | Reconnect.                                      |
```
Comment (oliwiapolec) [inline src/pages/messaging/customer-chat-api/v3.4/rtm-pushes/index.mdx:527]:
```suggestion
| `connection_evicted`                 | The Customer Chat API should be available again soon.                                                                                | Reconnect.                                      |
```

File: src/pages/messaging/customer-chat-api/v3.4/rtm-pushes/index.mdx [hunk 18]
```diff
@@ -519,22 +519,22 @@ Informs that a Customer was disconnected. The payload contains the reason of Cus
...
+| `license_expired`                    | The license has expired.                                                                                                             | Don't reconnect.                                |
+| `misdirected_connection` **\*\***    | Customer connected to a server in the wrong region.                                                                                  | Don't reconnect.                                |
+| `unsupported_version`                | Connecting to an unsupported version of the Customer Chat API.                                                                       | Don't reconnect.                                |
+| `product_version_changed`            | The product version has changed.                                                                                                     | Don't reconnect.                                |
+| `service_temporarily_unavailable`    | The Customer Chat API should be available again soon;                                                                                | Reconnect.                                      |
```
Comment (oliwiapolec) [inline src/pages/messaging/customer-chat-api/v3.4/rtm-pushes/index.mdx:535]:
```suggestion
| `service_temporarily_unavailable`    | The Customer Chat API should be available again soon.                                                                                | Reconnect.                                      |
```

File: src/pages/messaging/agent-chat-api/v3.1/rtm-pushes/index.mdx [hunk 19]
```diff
@@ -690,21 +690,22 @@ Informs that an Agent was disconnected. The payload contains the reason of Custo
...
+| `license_not_found`                 | License with the specified ID doesn't exist.                                                                                         | Don't reconnect.                                |
+| `misdirected_connection` **\***     | Agent connected to the server in the wrong region.                                                                                   | Don't reconnect.                                |
+| `unsupported_version`               | Connecting to an unsupported version of the Agent API.                                                                               | Don't reconnect.                                |
+| `product_version_changed`           | Product version has been changed.                                                                                                    | Don't reconnect.                                |
+| `too_many_connections`              | Agent reached the max. number of connections.                                                                                        | Reconnect.                                      |
```
[the remark this answers is not in the document]
Comment (grz-gajda) [reply src/pages/messaging/agent-chat-api/v3.1/rtm-pushes/index.mdx:707]: Oba są poprawne, od wersji 3.3 zmieniliśmy obsługę tego przypadku i dlatego nie zalecamy ponownego połączenia.

**Comment by grz-gajda** [comment]
@oliwiapolec I applied all the changes from the resolved discussions.

**Review by oliwiapolec — APPROVED** [review]

**Pull request merged**